gateway/mw_jwt.go

```diff
@@ -233,57 +232,97 @@ func (k *JWTMiddleware) getSecretToVerifySignature(r *http.Request, token *jwt.T
 	return []byte(session.JWTData.Secret), nil
 }
 
-func (k *JWTMiddleware) getSecretFromMultipleJWKURIs(jwkURIs []apidef.JWK, kid interface{}, keyType string) (interface{}, error) {
+func (k *JWTMiddleware) specCacheKey(spec *APISpec, prefix string) string {
+	return prefix + spec.APIID + spec.OrgID
+}
+
+func (k *JWTMiddleware) getSecretFromMultipleJWKURIs(jwkURIs []apidef.JWK, kidVal interface{}, keyType string) (interface{}, error) {
 	var (
-		wg       sync.WaitGroup
-		once     sync.Once
-		resultCh = make(chan interface{}, 1)
-		errCh    = make(chan error, len(jwkURIs))
+		jwkSets         []*jose.JSONWebKeySet
+		fallbackJWKURIs []apidef.JWK
+		kid, ok         = kidVal.(string)
 	)
 
-	for _, jwk := range jwkURIs {
-		wg.Add(1)
-		go func(uri string) {
-			defer wg.Done()
-			secret, err := k.getSecretFromURL(uri, kid, keyType)
-			if err == nil && secret != nil {
-				once.Do(func() {
-					resultCh <- secret
-				})
-			} else {
-				errCh <- fmt.Errorf("uri %s: %w", uri, err)
+	if !ok {
+		return nil, ErrKIDNotAString
+	}
+
+	cacheAPIDef := k.specCacheKey(k.Spec, "jwks_api_def_")
```

Comment:
NIT: recommend making this string value a constant jwkAPIPrefix or something like that, i assume it is also being used elsewhere.